sympy/functions/special/tests/test_bessel.py

```diff
@@ -81,11 +81,12 @@ def test_rewrite():
     # complex-valued, whereas SBFs are defined only for integer orders)
     order = nu
     for f in (besselj, bessely):
-        assert yn(order, z) == yn(order, z).rewrite(f)
-        assert jn(order, z) == jn(order, z).rewrite(f)
         assert hn1(order, z) == hn1(order, z).rewrite(f)
         assert hn2(order, z) == hn2(order, z).rewrite(f)
 
+    assert jn(order, z).rewrite(besselj) == sqrt(2)*sqrt(pi)*sqrt(1/z)*besselj(order + S(1)/2, z)/2
```

Comment:
order is a complex number (return value of randcmplx()), therefore .rewrite(besselj) must not perform a rewrite. see https://en.wikipedia.org/wiki/Bessel_function#Spherical_Bessel_functions:_jn.2C_yn